main.cpp

```diff
@@ -101,52 +135,116 @@ std::string Piano::displayChoosenSound(double soundKnobValue)
     return organSound;
 }
 
+double Piano::audio(int notePressed, double volume, bool isFootPedalPressed)
+{
+    if (isFootPedalPressed)
+    {
+        return 440.0 * volume;
+    }
+    return notePressed;
+}
+
 
 struct Microwave
 {
-    bool doorIsOpen = true;
-    double secondsRemaining = 0.0;
-    double power = 700.0;
-    int presetDishInput = 0;
+    Microwave();
+    bool doorIsOpen;
+    double secondsRemaining, power;
+    int presetDishInput;
 
     struct Preset
     {
-        float weightInGrams = 500.0f; 
-        int seconds = 1200;  
-        int power = 900; 
-        std::string presetName = "Potato";
+        float weightInGrams; 
+        int seconds;  
+        int power; 
+
+        Preset(float presetWeightInGrams,int presetSeconds,int presetPower);
```

Comment:
Nested classes need to have 5 properties and 3 member functions.
This class has 3 properties, and zero member functions, excluding the constructor.
- [ ] revise